### PR TITLE
Rollback Block During Processing

### PR DESCRIPTION
**Pull request by nisdas**
**What type of PR is this?**

Bug Fix

**What does this PR do? Why is it needed?**

In the event that there are issues during the processing of a block we rollback the block insertion so as to ensure that the node is not left in an inconsistent state. This will allow the node to be able to process the block quickly again via req/resp . 

**Which issues(s) does this PR fix?**

N.A

**Other notes for review**

**Acknowledgements**

- [x] I have read [CONTRIBUTING.md](https://github.com/prysmaticlabs/prysm/blob/develop/CONTRIBUTING.md).
- [x] I have made an appropriate entry to [CHANGELOG.md](https://github.com/prysmaticlabs/prysm/blob/develop/CHANGELOG.md).
- [x] I have added a description to this PR with sufficient context for reviewers to understand this PR.

### CODE REVIEW

**Review by potuz — DISMISSED** [review]
LGTM

File: beacon-chain/blockchain/process_block.go [hunk 1]
```diff
@@ -76,6 +76,7 @@ func (s *Service) postBlockProcess(cfg *postBlockProcessConfig) error {
 
 	err := s.cfg.ForkChoiceStore.InsertNode(ctx, cfg.postState, cfg.roblock)
 	if err != nil {
+		s.rollbackBlock(ctx, cfg.roblock.Root())
```
Comment (saolyn) [inline beacon-chain/blockchain/process_block.go:79]: Should we have a unit test triggering this?

**Review by potuz — DISMISSED** [review]

File: beacon-chain/blockchain/process_block_test.go [hunk 2]
```diff
@@ -2293,6 +2293,59 @@ func Test_getFCUArgs(t *testing.T) {
...
+	require.NoError(t, err)
+	require.Equal(t, true, hasState)
+
+	// Set invalid parent root to trigger forkchoice error.
+	wsb.SetParentRoot([]byte("bad"))
+	roblock, err := consensusblocks.NewROBlockWithRoot(wsb, root)
+	require.NoError(t, err)
+
+	// Rollback block insertion into db and caches.
+	require.ErrorContains(t, "invalid parent root", service.postBlockProcess(&postBlockProcessConfig{ctx, roblock, [32]byte{}, postState, false}))
```
Comment (potuz) [inline beacon-chain/blockchain/process_block_test.go:2340]: Could you check for "could not insert block" and "to fork choice store" instead?

**Review by potuz — APPROVED** [review]
TSM

**Pull request merged**